03-event-emitters/3-task/csv-reader.ts

```diff
@@ -0,0 +1,18 @@
+import * as fs from "node:fs";
+import csvtojson from "csvtojson";
+
+class CSVReader {
+  inputPath: string;
+
+  constructor(inputPath: string) {
+    this.inputPath = inputPath;
+  }
+
+  async read(): Promise<string[]> {
+    const fileStream = fs.createReadStream(this.inputPath);
+    const jsonObjects = await csvtojson().fromStream(fileStream);
+    return jsonObjects.map((obj) => JSON.stringify(obj));
```

Comment:
This implementation works, but it does not fully use the power of streams. Do you remember the notion of pipes? You can pipe streams together like so: readStream >> writeStream and there is no need to await anything. As soon as the read stream emits data, it will be forwarded to the writeStream. 

In your solution you await for the whole csv to be read instead of piping it on. 

Also, notice that streams output  buffers, and as we learned, buffer are bimary data which can be decoded, hence you can pass an encoding option to your createReadStream, which will convert your buffer into a string without the need to Json.stringify() it, something like this:

![image](https://github.com/molek95/nodejs-global-mentoring/assets/39033848/ba2da50f-4e10-4135-a06c-637b3cbb979c)